ios/PolyPodApp/PolyPod/AppDelegate.swift

```diff
@@ -21,6 +21,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Location tracking is disabled for now - no feature needs it
         //LocationTracker.shared.startLocationLogging()
         
+        UpdateNotification.instance.isProtectedDataAvailable = UIApplication.shared.isProtectedDataAvailable
```

Comment:
Maybe you should call `protectedDataDidBecomeAvailable` here? Isn't that what it was created for?

Reply:
I'm not a huge fan of making `UpdateNotification` a singleton over this - a singleton that can also be instantiated directly, that's a bit of a code smell in my mind :D How about simply adding a static var `isProtectedDataAvailable` to `UpdateNotification`? All instances would then access that one whenever the state is updated.

Reply:
`UpdateNotification` is made singleton not for the means of setting `isProtectedDataAvailable`. The whole idea is that the UpdateNotification would keep its state in memory, not requiring interaction with UserDefaults for all of its operations (which is problematic when protected data is not available). And this requires having a single instance.

The initialiser is available at the moment only for `testing` purpose. We have tests which require different UpdateNotification instance inside the same test. I may need to look on how to update those tests.

Reply:
I see, then I misread that :)